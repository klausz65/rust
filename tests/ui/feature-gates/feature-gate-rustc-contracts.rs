#![crate_type = "lib"]

#[core::contracts::requires(x > 0)]
pub fn requires_needs_it(x: i32) { }
//~^^  ERROR use of unstable library feature `rustc_contracts`
//~^^^ ERROR contracts are experimental

#[core::contracts::ensures(|ret| *ret > 0)]
pub fn ensures_needs_it() -> i32 { 10 }
//~^^  ERROR use of unstable library feature `rustc_contracts`
//~^^^ ERROR contracts are experimental
