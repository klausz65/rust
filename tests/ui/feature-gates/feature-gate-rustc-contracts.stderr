error[E0658]: use of unstable library feature `rustc_contracts`
  --> $DIR/feature-gate-rustc-contracts.rs:3:3
   |
LL | #[core::contracts::requires(x > 0)]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #133866 <https://github.com/rust-lang/rust/issues/133866> for more information
   = help: add `#![feature(rustc_contracts)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `rustc_contracts`
  --> $DIR/feature-gate-rustc-contracts.rs:8:3
   |
LL | #[core::contracts::ensures(|ret| *ret > 0)]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #133866 <https://github.com/rust-lang/rust/issues/133866> for more information
   = help: add `#![feature(rustc_contracts)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: contracts are experimental
  --> $DIR/feature-gate-rustc-contracts.rs:3:1
   |
LL | #[core::contracts::requires(x > 0)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #133866 <https://github.com/rust-lang/rust/issues/133866> for more information
   = help: add `#![feature(rustc_contracts)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: contracts are experimental
  --> $DIR/feature-gate-rustc-contracts.rs:8:1
   |
LL | #[core::contracts::ensures(|ret| *ret > 0)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #133866 <https://github.com/rust-lang/rust/issues/133866> for more information
   = help: add `#![feature(rustc_contracts)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0658`.
